### DISCUSSION TITLE
marp cannot be previewed in markdown preview enhanced

### DISCUSSION THREAD

**Discussion by yixing312** [marp-team/marp#423] [Q&A] [answered]
I'm sorry to have to ask such basic questions.There were some issues with preview when I was using marp.
While using  MPE:
![image](https://user-images.githubusercontent.com/96825143/226485152-8daf3cf1-1369-4f8d-b84a-bef82fd881b1.png)
It does not recognize page breaks,only with a line laying .
While using preview:
It will be like 
![image](https://user-images.githubusercontent.com/96825143/226485655-8341a04b-4252-4f89-9860-587cac53f8cd.png)

The version I'm currently using is
![image](https://user-images.githubusercontent.com/96825143/226485934-6ece40d5-9409-40f1-9ffa-0e0cbe194072.png)
![image](https://user-images.githubusercontent.com/96825143/226485962-c6fe310f-85ae-46f7-8a3c-3b19af25c429.png)

**Comment by yhatt** [comment, marked as answer]
It's important to know the difference between two extensions.

Marp for VS Code [extends the Markdown preview feature of VS Code](https://code.visualstudio.com/api/extension-guides/markdown-extension) to add Marp's preview feature. On the other hand, Markdown Preview Enhanced hijacks VS Code's Markdown preview feature, _and replaces it with its own preview feature that the extension has._ Especially the preview icon in the toolbar is replaced from VS Code's to Markdown Preview Enhanced's.

Since the preview feature provided by Markdown Preview Enhanced has not supported Marp, it cannot display the preview of Marp.

- https://github.com/shd101wyy/vscode-markdown-preview-enhanced/issues/1239
- https://github.com/shd101wyy/vscode-markdown-preview-enhanced/issues/596

To use Marp preview while using MPE, open the command palette via <kbd>Ctrl</kbd> (<kbd>Cmd</kbd>) + <kbd>Shift</kbd> + <kbd>P</kbd> and choose **"Markdown: Open Preview to the Side"** for VS Code built-in preview, instead of "Markdown: Markdown Preview Enhanced: Open Preview to the Side".

![](https://user-images.githubusercontent.com/3993388/226510460-bc635bc2-bb97-43c8-a4f8-502c2c0f8f00.png)

To restore a VS Code original preview button from the toolbar, disable **`markdown-preview-enhanced.hideDefaultVSCodeMarkdownPreviewButtons`** the MPE extension setting.

### Related issues

- https://github.com/marp-team/marp-vscode/issues/82
- https://github.com/marp-team/marp-vscode/issues/179#issuecomment-1062803781
- https://github.com/marp-team/marp-vscode/issues/336
- https://github.com/marp-team/marp-vscode/issues/459
